### PR TITLE
fix: [CDS-36920]: cache updated after service update, yaml getting deleted issue, ff updated

### PR DESCRIPTION
**Pull request by bhavya-sinha**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx [hunk 1]
```diff
@@ -325,14 +332,17 @@ export const EditStageView: React.FC<EditStageViewProps> = ({
...
+                    <>
+                      <div>
+                        <SelectDeploymentType
+                          viewContext={context}
+                          selectedDeploymentType={selectedDeploymentType}
+                          isReadonly={isReadonly}
+                          handleDeploymentTypeChange={handleDeploymentTypeChange}
+                        />
+                      </div>
+                      {/* <FormInput.CheckBox name="gitOpsEnabled" label={getString('common.gitOps')} /> */}
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployStage/EditStageView/EditStageView.tsx:344]: same

**Comment by bhavya-sinha** [comment]
retrigger titlecheck

**Comment by bhavya-sinha** [comment]
retrigger jest

File: src/modules/70-pipeline/components/ManifestSelection/ManifestSelection.tsx [hunk 2]
```diff
@@ -47,6 +47,7 @@ export default function ManifestSelection({
   const { getRBACErrorMessage } = useRBACError()
   const getServiceCacheId = `${pipeline.identifier}-${selectedStageId}-service`
   const { getCache } = useCache([getServiceCacheId])
+  const serviceInfo = getCache(getServiceCacheId) as ServiceDefinition
```
Comment (vkbansal) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestSelection.tsx:50]:
```suggestion
  const serviceInfo = getCache<ServiceDefinition>(getServiceCacheId)
```
This should work

File: src/modules/70-pipeline/components/WorkflowVariablesSelection/WorkflowVariables.tsx [hunk 3]
```diff
@@ -60,6 +60,7 @@ export default function WorkflowVariables({
   const [parentStageData, setParentStageData] = React.useState<StageElementWrapperConfig>()
   const getServiceCacheId = `${pipeline.identifier}-${selectedStageId}-service`
   const { getCache } = useCache([getServiceCacheId])
+  const serviceInfo = getCache(getServiceCacheId) as ServiceDefinition
```
Comment (vkbansal) [inline src/modules/70-pipeline/components/WorkflowVariablesSelection/WorkflowVariables.tsx:63]:
```suggestion
  const serviceInfo = getCache<ServiceDefinition>(getServiceCacheId)
```

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Pull request merged**